src/gateways/ElasticsearchGateway.ts

```diff
@@ -31,7 +27,30 @@ export class ElasticsearchGateway {
     await this.client.index({
       index: this.indexName,
       id: metadata.documentId,
-      body: metadata
+      body: metadata,
     });
   }
+
+  async findDocuments(metadata: DocumentMetadata): Promise<string[]> {
```

Comment:
I think `DocumentMetadata` has `documentId` as a required field, whereas I don't think this will be true of searching for documents. Probably worth a new interface for this. I think some of the other use cases define their own interface inside the file, have a look at the `IndexDocument` one for an example.

Reply:
Good shout!

Reply:
Looks nice now! Thanks!